Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py

```diff
@@ -305,15 +302,21 @@ def _get_self_deployed_token(self,
             return self._get_token_device_code(refresh_token, scope, integration_context)
         else:
             # by default, grant_type is CLIENT_CREDENTIALS
+            if self.multi_resource:
+                expires_in = -1  # init variable as an int
+                for resource in self.resources:
+                    self.resource = resource
+                    access_token, expires_in, refresh_token = self._get_self_deployed_token_client_credentials()
+                    self.resource_to_access_token[resource] = access_token
+                self.resource = ''
+                return '', expires_in, refresh_token
```

Comment:
The integration uses grant type of "client credentials" ( in the authorization process).
In order to use Key Vault endpoints, the integration need authorization to 'https://management.azure.com' resource.
But, in order to use Keys, Secrets and Certificates endpoints, the integration need authorization to 'https://vault.azure.net' resource.

**Bottom line** - in order to get authorization for all the endpoints, MicrosoftClient should initialize the access token for each resource (which mentioned earlier).
Since the "multi resource" implementation was missing for grant type of client credentials, I had to add it to MicrosoftClient. (I consulted with Bar Genish regards this issue).